FirebaseMLModelDownloader/Sources/LocalModelInfo.swift

```diff
@@ -58,22 +69,25 @@ struct ModelInfo {
       let path = defaults.value(forKey: "\(defaultsPrefix).model-path") as? String else {
       return nil
     }
-    name = modelName
-    self.downloadURL = url
-    self.modelHash = modelHash
-    self.size = size
-    self.path = path
+    self.init(name: name, downloadURL: url, modelHash: modelHash, size: size, path: path)
   }
 
-  func writeToDefaults(_ defaults: UserDefaults, appName: String) throws {
-    guard let modelPath = path else {
-      throw DownloadedModelError
-        .fileIOError(description: "Could not save model info to user defaults.")
-    }
-    let defaultsPrefix = ModelInfo.getUserDefaultsKeyPrefix(appName: appName, modelName: name)
+  /// Write local model info to user defaults.
+  func writeToDefaults(_ defaults: UserDefaults, appName: String) {
```

Comment:
nit: I'm not a big fan of mixing the data model and persistent storage into a single class mostly because of the following:
- it violates the "single responsibility" principle
- it's more difficult to write tests
- it's more difficult to update the persistent storage implementation in the future if needed.

It doesn't look critical to me here but we can discuss the alternatives if you are interested please let me know.

Reply:
Yes! Actually in a future PR, I broke out writeToDefaults as a required method of a protocol that LocalModelInfo conforms to. Is that more in line with what you had in mind?

Reply:
I'll take a look! Sorry, started from this PR and haven't with the next one yet.

Reply:
Ah sorry that PR is WIP and not ready for review yet. Would you like me to make that change here?

Reply:
A separate PR SGTM